src/org/labkey/remoteapi/assay/Protocol.java

```diff
@@ -0,0 +1,355 @@
+            _allowQCStates = (Boolean)json.get("allowQCStates");
+        if (json.containsKey("allowSpacesInPath"))
+            _allowSpacesInPath = (Boolean)json.get("allowSpacesInPath");
+        if (json.containsKey("allowTransformationScript"))
+            _allowTransformationScript = (Boolean)json.get("allowTransformationScript");
+        if (json.containsKey("autoCopyTargetContainerId"))
+            _autoCopyTargetContainerId = (String)json.get("autoCopyTargetContainerId");
+
+        if (json.get("availableDetectionMethods") instanceof JSONArray)
+        {
+            for (Object detectionMethod : (JSONArray)json.get("availableDetectionMethods"))
+                _availableDetectionMethods.add((String)detectionMethod);
+        }
+        if (json.containsKey("selectedDetectionMethod"))
+            _selectedDetectionMethod = (String)json.get("selectedDetectionMethod");
+
+        if (json.containsKey("availableMetadataInputFormats"))
+        {
+            _availableMetadataInputFormats = (HashMap<String,String>)json.get("availableMetadataInputFormats");
+        }
+        if (json.containsKey("selectedMetadataInputFormat"))
+            _selectedMetadataInputFormat = (String)json.get("selectedMetadataInputFormat");
+
+        if (json.get("availablePlateTemplates") instanceof JSONArray)
+        {
+            for (Object plateTemplate : (JSONArray)json.get("availablePlateTemplates"))
+                _availablePlateTemplates.add((String)plateTemplate);
+        }
+        if (json.containsKey("selectedPlateTemplate"))
+            _selectedPlateTemplate = (String)json.get("selectedPlateTemplate");
+
+        if (json.get("protocolTransformScripts") instanceof JSONArray)
+        {
+            for (Object transformScript : (JSONArray)json.get("protocolTransformScripts"))
+                _protocolTransformScripts.add((String)transformScript);
+        }
+        if (json.containsKey("protocolParameters"))
+        {
+            _protocolParameters = (HashMap<String,String>)json.get("protocolParameters");
+        }
+    }
+
+    public JSONObject toJSONObject()
+    {
+        JSONObject result = new JSONObject();
+        result.put("protocolId", _protocolId);
+        result.put("name", _name);
+        result.put("description", _description);
+        result.put("providerName", _providerName);
+        JSONArray domains = new JSONArray();
+        result.put("domains", domains);
+        for (Domain domain : _domains)
+            domains.add(domain.toJSONObject(true));
+
+
+        if (_backgroundUpload != null)
+            result.put("backgroundUpload", _backgroundUpload);
+
+        if (_editableResults != null)
+            result.put("editableResults", _editableResults);
+
+        if (_qcEnabled != null)
+            result.put("qcEnabled", _qcEnabled);
+
+        if (_allowTransformationScript != null)
+            result.put("allowTransformationScript", _allowTransformationScript);
```

Comment:
I believe you want the `protocolTransformScripts` prop here in the result instead of the `allowTransformationScript `, right?

Reply:
right